fix: check for Virtual Machine only for WSL

jeffmaury opened the pull request:
Fixes #9177
Fixes #9610

### What does this PR do?

Check for VirtualMachine presence only for WSL

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

#9177 

### How to test this PR?

1. Remove podman
2. Virtual Machine
3. Check that preflights are ok

- [x] Tests are covering the bug fix or the new feature

benoitf reviewed (commented):
codewise LGTM but I'm on macOS

odockal commented:
I would like to verify on Windows after testing.

odockal reviewed (approved):
From the issue perspective, we can now install podman when having a hyperv only enabled. I have opened a follow up issue to tackle other problem I have discovered: https://github.com/containers/podman-desktop/issues/9610

odockal reviewed (changes requested):
Sorry, I have changed my mind: I have tried again and specified what will happen if we will merge this issue: https://github.com/containers/podman-desktop/issues/9610. 
Basically, now we would be forcing user to open pd with admin right when he has no hyperv or WSL enabled yet, just to tell him he needs hyperV. We need to improve the workflow when doing requirements check.

jeffmaury commented:
> Sorry, I have changed my mind: I have tried again and specified what will happen if we will merge this issue: #9610. Basically, now we would be forcing user to open pd with admin right when he has no hyperv or WSL enabled yet, just to tell him he needs hyperV. We need to improve the workflow when doing requirements check.

I updated so that we return the first check that failed (WSL in that case). It needs another follow up PR as we should inform that both WSL and HyperV are not valid but needs discussion so not for 1.14
